sdk/src/main/java/co/omise/android/ui/CreditCardActivity.kt

```diff
@@ -0,0 +1,154 @@
+package co.omise.android.ui
+
+import android.app.Activity
+import android.content.Intent
+import android.os.Bundle
+import android.view.View
+import android.widget.Button
+import android.widget.TextView
+import androidx.appcompat.app.AppCompatActivity
+import co.omise.android.R
+import co.omise.android.api.Client
+import co.omise.android.api.RequestListener
+import co.omise.android.extensions.setOnAfterTextChangeListener
+import co.omise.android.extensions.setOnClickListener
+import co.omise.android.models.APIError
+import co.omise.android.models.Token
+import kotlinx.android.synthetic.main.activity_credit_card.button_submit
+import kotlinx.android.synthetic.main.activity_credit_card.edit_card_name
+import kotlinx.android.synthetic.main.activity_credit_card.edit_card_number
+import kotlinx.android.synthetic.main.activity_credit_card.edit_expiry_date
+import kotlinx.android.synthetic.main.activity_credit_card.edit_security_code
+import kotlinx.android.synthetic.main.activity_credit_card.text_error_message
+import java.io.IOError
+
+class CreditCardActivity : AppCompatActivity() {
+
+    private val cardNumberEdit: CreditCardEditText by lazy { edit_card_number }
+    private val cardNameEdit: CardNameEditText by lazy { edit_card_name }
+    private val expiryDateEdit: ExpiryDateEditText by lazy { edit_expiry_date }
+    private val securityCodeEdit: SecurityCodeEditText by lazy { edit_security_code }
+    private val submitButton: Button by lazy { button_submit }
+    private val errorMessageText: TextView by lazy { text_error_message }
+
+    private val editTexts: List<OmiseEditText> by lazy {
+        listOf(cardNumberEdit, cardNameEdit, expiryDateEdit, securityCodeEdit)
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        if (!intent.hasExtra(EXTRA_PKEY)) {
+            throw IllegalAccessException("Can not found ${::EXTRA_PKEY.name}.")
+        }
+
+        setContentView(R.layout.activity_credit_card)
+        setTitle(R.string.default_form_title)
+
+        submitButton.setOnClickListener(::submit)
+
+        editTexts.forEach {
+            it.setOnFocusChangeListener { _, hasFocus ->
+                if (!hasFocus) {
+                    try {
+                        it.validate()
+                    } catch (e: InputValidationException.InvalidInputException) {
+                        it.errorMessage = getString(R.string.error_invalid, it.hint)
+                    } catch (e: InputValidationException.EmptyInputException) {
+                        it.errorMessage = null
+                    }
+                } else {
+                    it.errorMessage = null
+                }
+            }
+
+            it.setOnAfterTextChangeListener(::updateSubmitButton)
+        }
+    }
+
+    override fun onBackPressed() {
+        setResult(RESULT_CANCELED)
+        super.onBackPressed()
+    }
+
+    private inner class ActivityRequestListener : RequestListener<Token> {
```

Comment:
Perhaps we can use a better name here? `CreditCardRequestListener` or just `listener`?

Reply:
Yeah, I'll change this.